src/main/scala/spark/dicom/deidentifier/DicomDeidentifier.scala

```diff
@@ -0,0 +1,84 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package ai.kaiko.spark.dicom.deidentifier
+
+import ai.kaiko.dicom.DicomDeidentifyDictionary
+import ai.kaiko.dicom.DicomStandardDictionary
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions._
+import org.dcm4che3.data._
+
+
+case class DicomDeidAction(
+  keyword: String,
+  vr: VR,
+  action: String
+)
+
+object DicomDeidentifier {
+
+  def deidentify(dataframe: DataFrame): DataFrame = {
+
+    val deidActions: Array[DicomDeidAction] = DicomDeidentifyDictionary.elements.flatMap(
+      deidElem => DicomStandardDictionary.keywordMap.get(deidElem.keyword) match {
+        case Some(stdElem) => stdElem.vr match {
+          case Right(vr) => Some(DicomDeidAction(deidElem.keyword, vr, deidElem.action))
+          case _ => None
+        }
+        case _ => None
+      }
+    )
+
+    val deidActionGroups = deidActions.groupBy(_.action)
+
+    val toKeep = deidActionGroups.getOrElse("K", Array())
+    val toEmpty = deidActionGroups.getOrElse("Z", Array()) ++ deidActionGroups.getOrElse("Z/D", Array())
```

Comment:
I'd probably create an overarching data structure instead of multiple variables to make code more composable